draft-ietf-httpbis-semantics-latest.xml

```diff
@@ -1711,6 +1711,78 @@ deployed and not likely to be registered in a timely manner otherwise.</t>
 </section>
 </section>
 
+<section title="Trailer Fields" anchor="trailer.fields">
+   <iref item="trailer fields"/>
+   <iref item="trailers"/>
+
+<section title="Purpose" anchor="trailers.purpose">
+<t>
+   In some cases specific to the transfer coding or HTTP version, additional
+   metadata can be sent after the initial header section has been completed
+   (during or after transmission of the payload body) that might be
+   dynamically generated while the body is being sent, such as a message
+   integrity check, digital signature, or post-processing status.
+   For example, the chunked coding in HTTP/1.1 allows a trailer section after
+   the payload body (<xref target="chunked.trailer.section"/>) which can contain
+   trailer fields: field names and values that share the same syntax and
+   namespace as header fields but are received after the header section.
+</t>
+<t>
+   Trailer fields ought to be processed and stored separately from the fields
+   in the header section to avoid overriding the message semantics known at
+   the time the header section was complete. The presence or absence of
+   certain header fields might impact choices made for the routing or
+   processing of the message as a whole before the trailers are received;
+   those choices cannot be unmade by the later discovery of trailer fields.
+</t>
+</section>
+
+<section title="Limitations" anchor="trailers.limitations">
+<t>
+   Many header fields cannot be processed outside the header section because
+   their evaluation is necessary prior to receiving the message body, such as
+   fields that describe message framing, routing, authentication,
+   request modifiers, response controls, or payload format.
+   A sender &MUST-NOT; generate a trailer field unless the sender knows the
+   corresponding header field name's definition permits the field to be sent
+   in trailers.
+</t>
+<t>
+   Because of the potential for trailer fields to be discarded in transit, a
```

Comment:
It feels like we should say a little bit more about this; right now we're relying on the reader to infer that some HTTP versions don't support trailers, that intermediaries can drop them, and that some implementations just drop them on the floor. Probably worth including in "purpose" above.